File: Arrays/Two Pointer/TwoPointer.cs
// Reverse Words in a String
// Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order
// LeetCode 557.

public class Solution
{
    public string ReverseWords(string s)
    {
        string[] words = s.Split(" ");

        for (int i = 0; i < words.Length; i++)
        {
            char[] characters = words[i].ToCharArray();
            int l = 0;
            int h = characters.Length - 1;
            while (l < h)
            {
                (characters[l], characters[h]) = (characters[h], characters[l]);
                l++;
                h--;
            }
            words[i] = new string(characters);
        }
        return String.Join(" ", words);
    }

}

File: Arrays/Two Pointer/two_pointer.py
#Reverse Words in a String
# Given a string s, reverse the order of characters in each word within a sentence while still preserving whitespace and initial word order
# LeetCode 557.
class Solution:
    def reverseWords_manual(s):
        res = ''
        l, r = 0,0

        while r < len(s):
            if s[r] != ' ':
                r += 1
            else:
                res += s[l:r+1][::-1]
                r += 1
                l = r

        res += ' '
        res += s[l:r + 2][::-1]
        return res[1:]